code/game/objects/items/mail.dm

```diff
@@ -332,3 +352,172 @@
+	user.temporarilyRemoveItemFromInventory(src, force = TRUE)
+	playsound(loc, 'sound/items/poster_ripped.ogg', vol = 50, vary = TRUE)
+	for(var/obj/item/stuff in contents) // Mail and envelope actually can have more than 1 item.
+		if(user.put_in_hands(stuff) && armed)
+			log_bomber(user, "opened armed mail made by [made_by_cached_name] ([made_by_cached_ckey]), activating", stuff)
+			INVOKE_ASYNC(stuff, TYPE_PROC_REF(/obj/item/mail/traitor, attack_self), user)
+	qdel(src)
+
+/obj/item/mail/traitor/multitool_act(mob/living/user, obj/item/tool)
+	if(armed == FALSE || user.get_inactive_held_item() != src)
+		. = ..()
+	else
+		if(user.mind == made_by.resolve())
+			balloon_alert(user, "disarming trap...")
+			if(!do_after(user, 2 SECONDS, target = src))
+				return FALSE
+			balloon_alert(user, "disarmed")
+			playsound(src, 'sound/machines/defib_ready.ogg', vol = 100, vary = TRUE)
+			armed = FALSE
+			return TRUE
+		else
+			balloon_alert(user, "tinkering with something...")
+			
+			if(!do_after(user, 2 SECONDS, target = src))
+				after_unwrap(user)
+				return FALSE
+			if(prob(50))
+				balloon_alert(user, "disarmed something...?")
+				playsound(src, 'sound/machines/defib_ready.ogg', vol = 100, vary = TRUE)
+				armed = FALSE
+				return TRUE
+			else
+				after_unwrap(user)
+				return TRUE
+
+/obj/item/storage/mail_counterfeit_device
+	name = "GLA-2 mail counterfeit device"
+	desc = "Device that actually able to counterfeit NT's mail. This device also able to place a trap inside of mail for malicious actions. Trap will \"activate\" any item inside of mail. Also it might be used for contraband purposes. Integrated micro-computer will give you great configuration optionality for your needs."
+	w_class = WEIGHT_CLASS_NORMAL
+	icon = 'icons/obj/device_syndie.dmi'
+	icon_state = "mail_counterfeit_device"
+
+/obj/item/storage/mail_counterfeit_device/Initialize(mapload)
+	. = ..()
+	atom_storage.max_slots = 1
+	atom_storage.allow_big_nesting = TRUE
+	atom_storage.max_specific_storage = WEIGHT_CLASS_NORMAL
+
+/obj/item/storage/mail_counterfeit_device/examine_more(mob/user)
+	. = ..()
+	var/list/msg = list()
+	msg += "[span_notice("<i>You notice the manufacture marking on the side of the device...</i>")]"
+	msg += "\t[span_info("Guerilla Letter Assembler")]"
+	msg += "\t[span_info("GLA Postal Service, right on schedule.")]"
+	return msg
+
+/obj/item/storage/mail_counterfeit_device/attack_self(mob/user, modifiers)
+	var/mail_type = tgui_alert(user, "Is it gonna be an envelope or a normal mail?", "Mail Counterfeiting", list("Mail", "Envelope"))
+	if(isnull(mail_type))
+		return FALSE
+	if(loc != user)
+		return FALSE
+	mail_type = lowertext(mail_type)
+
+	var/mail_armed = tgui_alert(user, "Is it gonna be armed?", "Mail Counterfeiting", list("Yes", "No"))
+	if(isnull(mail_armed))
+		return FALSE
+	if(loc != user)
+		return FALSE
+
+	if(mail_armed == "Yes")
+		mail_armed = TRUE
+	else
+		mail_armed = FALSE
```

Comment:
You didn't do my feedback here

Reply:
Ah, i see, i can change it just to
`shady_mail.armed = (mail_armed == "Yes")`